Sources/MapLibreSwiftUI/MapViewCoordinator.swift

```diff
@@ -0,0 +1,194 @@
+                } else {
+                    print("That's funny... couldn't find identifier \(sourceID)")
+                }
+            }
+        }
+
+        // Snapshot the new user-defined layers
+        snapshotUserLayers = parent.userLayers
+
+        // If the style is loaded, add the new layers to it.
+        // Otherwise, this will get invoked automatically by the style didFinishLoading callback
+        if let style = mapView.style {
+            addLayers(to: style)
+        }
+    }
+
+    func addLayers(to mglStyle: MLNStyle) {
+        for layerSpec in parent.userLayers {
+            // DISCUSS: What preventions should we try to put in place against the user accidentally adding the same layer twice?
+            let newLayer = layerSpec.makeStyleLayer(style: mglStyle).makeMLNStyleLayer()
+
+            // Unconditionally transfer the common properties
+            newLayer.isVisible = layerSpec.isVisible
+
+            if let minZoom = layerSpec.minimumZoomLevel {
+                newLayer.minimumZoomLevel = minZoom
+            }
+
+            if let maxZoom = layerSpec.maximumZoomLevel {
+                newLayer.maximumZoomLevel = maxZoom
+            }
+
+            switch layerSpec.insertionPosition {
+            case .above(layerID: let id):
+                if let layer = mglStyle.layer(withIdentifier: id) {
+                    mglStyle.insertLayer(newLayer, above: layer)
+                } else {
+                    NSLog("Failed to find layer with ID \(id). Adding layer on top.")
+                    mglStyle.addLayer(newLayer)
+                }
+            case .below(layerID: let id):
+                if let layer = mglStyle.layer(withIdentifier: id) {
+                    mglStyle.insertLayer(newLayer, below: layer)
+                } else {
+                    NSLog("Failed to find layer with ID \(id). Adding layer on top.")
+                    mglStyle.addLayer(newLayer)
+                }
+            case .aboveOthers:
+                mglStyle.addLayer(newLayer)
+            case .belowOthers:
+                mglStyle.insertLayer(newLayer, at: 0)
+            }
+        }
+    }
+}
+
+// MARK: - MLNMapViewDelegate
+
+extension MapViewCoordinator: MLNMapViewDelegate {
+    
+    public func mapView(_ mapView: MLNMapView, didFinishLoading mglStyle: MLNStyle) {
+        addLayers(to: mglStyle)
+    }
+
+    /// The MapView's region has changed with a specific reason.
+    public func mapView(_ mapView: MLNMapView, regionDidChangeWith reason: MLNCameraChangeReason, animated: Bool) {
+        let isFollowing = parent.camera.state == .trackingUserLocation && mapView.userTrackingMode == .follow
+        let isFollowingHeading = parent.camera.state == .trackingUserLocationWithHeading && mapView.userTrackingMode == .followWithHeading
+        let isFollowingCourse = parent.camera.state == .trackingUserLocationWithCourse && mapView.userTrackingMode == .followWithCourse
```

Comment:
Discussion points for the meeting:

* I see we're checking both the state value and the underlying map view property... I'm assuming it's necessary since you did it ;) Is it possible for these to get out of sync? It looks pretty OK as-is, but just trying to think through any edge cases just in case....
* Maybe these could be named a bit more descriptively

Reply:
Discussed in meeting: "describe this MapLibre witchcraft." Basically this avoids jank when the user interacts with the map during tracking and allows it to break out cleanly.